### ISSUE TITLE
2019.09 release planning

### ISSUE THREAD

**Issue by vit9696** [acidanthera/bugtracker#453]
I am not positive whether we should make a big release in September, but most likely some bugfixes should land before 10.15 release. Most importantly it is about the time to transition OpenCore to public beta. Below I listed some issues that we should try to focus on for this release, from most to least important.

- [x] AppleALC 1.4.1
- [x] AppleSupport 2.1.0
- [x] gfxutil 1.79b
- [x] HibernationFixup 1.2.8
- [x] IntelMausi 1.0.2
- [x] MacInfoPkg 2.0.7
    - [x] Ideally we add model information for new MBP15,3 and MBP15,4
- [x] OpenCore 0.5.0
    - [x] Definitely need to resolve https://github.com/acidanthera/bugtracker/issues/448 (https://github.com/acidanthera/OcSupportPkg/pull/8)
    - [x] At least a workaround for https://github.com/acidanthera/bugtracker/issues/445 
    - [x] Implementing MMIO devirtualisation quirk for Apple boot
    - [ ] Serial initialisation debug option (`Misc` → `SerialInit`) for boards not supporting it. Can take from EDK II.
    - [ ] Adding booter patches based on OcAppleBootCompat (https://github.com/acidanthera/bugtracker/issues/380)
    - [x] Make AppleUiSupport a set of OcSupportPkg libraries and merge into OpenCore. Also update AppleSupportPkg if we make it in time. https://github.com/acidanthera/bugtracker/issues/336
    - [ ] Improve `AvoidHighAlloc` https://github.com/acidanthera/bugtracker/issues/449
    - [x] Add NTFS and EXT filesystems to `ScanPolicy` since we support `BlessOverride` from now on.
- [x] VoodooPS2 2.0.3 (intermediate release for ~2019.09.19) 
    - [x] Trackpoint support https://github.com/acidanthera/bugtracker/issues/302#issuecomment-520162931
    - [x] Bugfixes if any
- [ ] VoodooPS2 2.1.0 (full release)
    - [ ] Merge with VoodooInput interface https://github.com/acidanthera/bugtracker/issues/329
- [x] WhateverGreen 1.3.2

We have three big tasks https://github.com/acidanthera/bugtracker/issues/385, https://github.com/acidanthera/bugtracker/issues/368, and https://github.com/acidanthera/bugtracker/issues/323, which need a lot of attention, but I do not think we should focus on them for september.
 
Issues to be taken action upon:
- @corpnewt, @kylon, @notiflux, since we are getting close to beta, September-October will be about the time for user apps to be in shape https://github.com/acidanthera/bugtracker/issues/330.
- Close or investigate whether https://github.com/acidanthera/bugtracker/issues/377 needs our attention
- Make decisions on guides

Side note, from that point Acidanthera will no longer officially support Clover bootloader. Things to keep in mind:
- We will not intentionally break anything that currently works and our code remains fully opensource and available for external contribution.
- We will no longer perform Clover compatibility release validation for our products, but we will fix our own bugs upon request.
- With future releases we may abandon select products, hacks specific to Clover, and Clover patch contribution to ease maintenance burden.

**Comment by Snikii** [comment]
@vit9696 since this upcoming update seems to be planned as a bigger one considering the fact that it will aim as a Public Beta release, i wanted to discuss about some of the concerns i have about Laptops compatibility in general.

First of all i do plan to make a project like RehabMan: [https://github.com/RehabMan/OS-X-Clover-Laptop-Config](url)
But of course for OpenCore: **macOS-OpenCore-Laptop-Config** or something named like that where i plan to list all available/possibly needed patches and some template SSDTs (pretty much similar project but for OpenCore.)

My concerns are related towards the missing features like TgtBridge and InfoPlistPatch in general.
You did explain/answer my question on InsanelyMac with some of your reasons why you don't plan to implement such features to OC and some of the alternatives that you prefer.
But honestly i still think Patching DSDT is not so relevant due to the fact that a minimal change into bios settings or hardware upgrade may change any of the OperationRegion (xxx, SystemMemory) which leads to a broken DSDT and will require to re-dump, re-extract and re-patch DSDT which is a PITA.

The patching features are mostly fine, we just need a "TgtBridge" patching and "InfoPlistPatch" feature.
I still think there are some important use cases for these especially on Laptops as they require more complex patches for example:
- Battery Status
- I2C Touchpads and many other cases.
I know that we have the "Skip", "Count" and "Limit" function but these are more technical and we don't have a very well documented use case of these as far as im aware of/checked.
Also these functions are not really that much optimal for guides in general (versatility).
As a bonus these patches also do not touch any of the operationRegions (SystemMemory).

We all know that even exact same laptop/Desktop may have slightly different ACPI Tables/structure either due to a different bios version/update or specific different hardware/part that may break the use of these functions if let's say we make a guide for a specific hardware.
The user that comes to the guide and grab the config.plist and the addon SSDTs, he may have slightly different Bios Update version or Slightly different hardware, the specific "X","Y" or "Z" patch that is done with "Skip","Count" or "Limit" may break for him/her.
While on the other hand even if a "TgtBridge" patch turns out to be wrong or not needed for him/her, it will be simply ignored if no such occurrence is found.

I do believe that it will be a much more friendly solution for everyone in general, especially for the people who transition from Clover to OpenCore and will be more versatile for same laptop/desktop models especially for guides in general.

Also as i read on the notes, Clover will not be officially supported anymore, everyone may eventually need to jump to OC and having these options will be so much easier.
Let it be a personal issue on how and how good everyone decides to optimize their hacks.
Better a working non efficient way rather that a non working solution.

Sorry for the TL:DR post.

Looking forward for the great work that all of you do and have done.
"Waiting patiently for that juicy secure boot compatibility"

Sincerely,
Sniki

**Comment by vit9696** [comment]
@sniki,

1. Please create a dedicated issue for "TgtBridge" just like I asked you [here](https://www.insanelymac.com/forum/topic/338516-opencore-discussion/?page=40&tab=comments#comment-2682248). I completely forgot about the thing as it does not affect anybody here. I can imagine something like that being added, in case we see justified need for this.
2. As for plist patches, they are still not planed. While I will not give you false hopes, as nobody here even has a laptop with I2C, but we are currently working with @ben9923 to fix VoodooI2C bugs requiring causing the need to patch Apple's I2C driver. Even if nothing is found we can always patch its probe method through OpenCore symbolic kext patcher.

**Comment by Andrey1970AppleLife** [comment]
"Ideally we add model information for new MBP15,3 and MBP15,4"
And for new MBA8,2

**Comment by vit9696** [comment]
Taking all things into consideration I conclude that we are release ready. @vandroiy2013, please handle that.
